run_project_tests.py

```diff
@@ -264,22 +264,22 @@ def __lt__(self, other: object) -> bool:
 failing_logs: T.List[str] = []
 print_debug = 'MESON_PRINT_TEST_OUTPUT' in os.environ
 under_ci = 'CI' in os.environ
-skip_scientific = under_ci and ('SKIP_SCIENTIFIC' in os.environ)
+ci_jobname = os.environ.get('MESON_CI_JOBNAME', None)
 do_debug = under_ci or print_debug
 no_meson_log_msg = 'No meson-log.txt found.'
 
-host_c_compiler: T.Optional[str]   = None
-compiler_id_map: T.Dict[str, str]  = {}
-tool_vers_map:   T.Dict[str, str]  = {}
+host_c_compiler: T.Optional[str] = None
+compiler_id_map: T.Dict[str, str] = {}
+tool_vers_map:   T.Dict[str, str] = {}
 
-compile_commands:   T.List[str]
-clean_commands:     T.List[str]
-test_commands:      T.List[str]
-install_commands:   T.List[str]
-uninstall_commands: T.List[str]
+compile_commands:   T.List[str] = []
+clean_commands:     T.List[str] = []
+test_commands:      T.List[str] = []
+install_commands:   T.List[str] = []
+uninstall_commands: T.List[str] = []
 
-backend:      'Backend'
-backend_flags: T.List[str]
+backend:       Backend = Backend.ninja
+backend_flags: T.List[str] = []
```

Comment:
Are these changes really required or automatically generated?

Reply:
The initializers are needed to prevent flake8 from reporting `F821 undefined name` (which on reflection, I guess is a bug in flake8, since the name should be defined by type-annotated name, even without an initial value)

I'm not sure the current annotation for the `backend` global is actually right:  the string `backend` isn't the type `backend`?

Maybe I should just drop the flake8 fixes commit (since we don't seem to be running flake8/sider anymore)

Reply:
I would say that we drop the changes then.

> I'm not sure the current annotation for the backend global is actually right: the string 'backend' isn't the type backend?

The `'Backend'` is a valid type annotation. It is a string so that `Backend` does not have to be imported outside the `if T.TYPE_CHECKING:` block.